langserver/handler.go

```diff
@@ -0,0 +1,65 @@
+package langserver
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/haya14busa/go-vimlparser"
+	"github.com/haya14busa/go-vimlparser/ast"
+
+	"github.com/sourcegraph/jsonrpc2"
+)
+
+func NewHandler() jsonrpc2.Handler {
+	var langHandler = &LangHandler{
+		files: make(map[string]*vimfile),
+	}
+	return jsonrpc2.HandlerWithError(langHandler.handle)
+}
+
+type LangHandler struct {
+	files map[string]*vimfile
+}
+
+type vimfile struct {
+	TextDocumentItem
+	Ast      *ast.File
+	AstError error
+}
+
+func NewVimFile(textDocumentItem TextDocumentItem) (result *vimfile, error error) {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported func NewVimFile returns unexported type *langserver.vimfile, which can be annoying to use